packages/InlineEditing/stories/InlineEditing.stories.mdx

```diff
@@ -0,0 +1,278 @@
+  options: {
+    isToolshown: true,
+    showPanel: false,
+  },
+};
+
+<Meta title={`${getStoryName("InlineEditing")}/Docs`} parameters={params} />
+
+<Toast kind={Toast.types.kind.WARNING} hasCloseButton={false}>
+  Work in progress ahead, this component still requires some more work to be ready for production, but the API is
+  stable, so you can start integrating it into your application.
+</Toast>
+
+# Inline Editing
+
+The `<InlineEditing />` component allows you to display two values for the user one when the componet is not in editing mode and an input whenever the user interact and set
+the component into editing mode.
+
+## How does it works?
+
+The InlineEditing package includes a component name `<Editor />` that you can use to encapsulated this kind of interaction. The `<Editor />` Component comes with two
+subcomponent `<Editor.Value />` and `<Editor.Edit />`. This two component will switch between each other whenever the prop `isEditing` on the `Editor` component chages.
+
+The following is an example of usign the Editor component to build a simple Text (value) `->` Text (input) interaction:
+
+<Blocks>
+  <Block>
+    <AppInlineInput />
+  </Block>
+  <Block>
+
+```jsx
+function InlineInput(props) {
+  const { value, onSubmit, isEditing, onClick } = props;
+  const refInput = React.useRef(null);
+
+  function handleClick() {
+    onClick();
+  }
+
+  function submit() {
+    onSubmit({ nextValue: refInput.current.value });
+  }
+
+  return (
+    <Editor onEdit={() => {}} isEditing={isEditing} onClick={handleClick}>
+      <Editor.Edit>
+        {/** we are going to only report back when the user hits Enter */}
+        <Input
+          defaultValue={value}
+          type="text"
+          ref={refInput}
+          onKeyUp={event => {
+            if (event.key === "Enter") {
+              submit();
+            }
+          }}
+        />
+      </Editor.Edit>
+      <Editor.Value>
+        <span style={{ textDecoration: "underline" }}>{value}</span>
+      </Editor.Value>
+    </Editor>
+  );
+}
+
+export function App() {
+  const [value, setValue] = React.useState("Click on it to edit it");
+  const [isEditing, setIsEditing] = React.useState(false);
+
+  function handleClick() {
+    setIsEditing(true);
+  }
+
+  function handleSubmit({ nextValue }) {
+    setValue(nextValue);
+    setIsEditing(false);
+  }
+
+  // prettier-ignore
+  return (
+    <InlineInput 
+      value={value} 
+      onClick={handleClick} 
+      isEditing={isEditing} 
+      onSubmit={handleSubmit} 
+    />
+  )
+}
+```
+
+  </Block>
+</Blocks>
+
+The previous example is a simple attempt to create a InlineEditng Input, to have it fully functionable will still required more work with the keyboard interactions,
+add focus onClick, cancel it onKeyUp === "Escape", etc.
+
+An because wiring a fully functionally InlineEditing component can be tedious and tricky to make it work properly, we currently are providing three components ready to use:
```

Comment:
```suggestion
And because wiring a fully functionally InlineEditing component can be tedious and tricky to make it work properly, we are currently providing three components ready to use:
```